PubNub/Network/Parsers/PNSubscribeParser.m

```diff
@@ -152,72 +166,54 @@ + (NSDictionary *)parsedServiceResponse:(id)response withData:(NSDictionary *)ad
     // through 'nil' initialized local variable.
     NSDictionary *processedResponse = nil;
     
-    // Array will arrive in case of subscription event
-    if ([response isKindOfClass:[NSArray class]]) {
-        
-        NSArray *feedEvents = response[kPNEventsListElementIndex];
-        NSNumber *timeToken = @([response[kPNEventTimeTokenElement] longLongValue]);
-        NSArray *channels = nil;
-        NSArray *groups = nil;
-        if ([(NSArray *)response count] > kPNEventChannelsElementIndex) {
-            
-            channels = [PNChannel namesFromRequest:response[kPNEventChannelsElementIndex]];
-        }
-        if ([(NSArray *)response count] > kPNEventChannelsDetailsElementIndex) {
-            
-            groups = [PNChannel namesFromRequest:response[kPNEventChannelsDetailsElementIndex]];
-        }
-        
-        // Checking whether at least one event arrived or not.
+    // response is subscription
+    if ([response isKindOfClass:[NSDictionary class]]) {
+        NSAssert(response[kPNEventsTimeTokenElementKey], @"There should always be a time token object for this key value");
+        NSDictionary *timeTokenDictionary = response[kPNEventsTimeTokenElementKey];
+        NSNumber *timeToken = @([timeTokenDictionary[kPNEventsTimeTokenElementKey] longLongValue]);
+        NSNumber *region = @([timeTokenDictionary[kPNEventsTimeTokenRegionElementKey] integerValue]);
+        NSArray *feedEvents = response[kPNEventsMessageElementKey];
         if ([feedEvents count]) {
-            
             NSMutableArray *events = [[NSMutableArray alloc] initWithCapacity:[feedEvents count]];
             for (NSUInteger eventIdx = 0; eventIdx < [feedEvents count]; eventIdx++) {
                 
                 // Fetching remote data object name on which event fired.
-                NSString *objectOrGroupName = (eventIdx < [channels count] ? channels[eventIdx] : channels[0]);
-                NSString *objectName = ([groups count] > eventIdx ? groups[eventIdx] : nil);
-                NSMutableDictionary *event = [self eventFromData:feedEvents[eventIdx]
-                                                      forChannel:(objectName?: objectOrGroupName)
-                                                           group:(objectName? objectOrGroupName: nil)
-                                        withAdditionalParserData:additionalData];
+                NSMutableDictionary *event = [self eventFromData:feedEvents[eventIdx] withAdditionalParserData:additionalData];
                 event[@"timetoken"] = timeToken;
+                event[@"region"] = region;
```

Comment:
@jzucker2 _event[@"timetoken"]_ and _event[@"region"]_ shouldn't be changed to global, if values for it has been passed in data stored under _feedEvents[eventIdx]_. There is key-path to concrete object publish data (_p.t_ and _p.r_). Only presence events has a bit different information about event (under _timetoken_ inside of _d_ object).